.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,21 @@
+name: pr-validation
+
+on:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-dotnet@v2
+      with:
+        global-json-file: global.json
+    - name: Install dependencies
+      run: dotnet restore
+    - name: Build
+      run: dotnet build dirs.proj
+    - name: Test with the dotnet CLI
+      run: dotnet test
```

Comment:
You may not need `dotnet build` or `dotnet restore` at all. I think `dotnet test` implicitly calls build (which implicitly calls restore).